src/core_plugins/kibana/public/management/sections/indices/create_index_pattern_wizard/step_index_pattern/append_wildcard.js

```diff
@@ -0,0 +1,53 @@
+import { uiModules } from 'ui/modules';
+
+const module = uiModules.get('apps/management');
+
+/**
+ * This directive automatically appends a wildcard to the input field
+ * after the user starts typing. It lets the user delete the wildcard
+ * if necessary. If the value of the input field is set back to an empty
+ * string, the wildcard is immediately re-appended after the user starts
+ * typing. This is intended to be a UX improvement for the index pattern
+ * creation page. See https://github.com/elastic/kibana/pull/13454
+ */
+module.directive('appendWildcard', function () {
+  return {
+    require: 'ngModel',
+    link: function ($scope, $elem, $attrs, $ctrl) {
+      $scope.appendedWildcard = false;
+
+      function setElementValue(val, event, moveSelectionRange = false) {
+        event.preventDefault();
+        $elem.val(val);
+        if (moveSelectionRange) {
+          $elem[0].setSelectionRange(1, 1);
+        }
+      }
+
+      $elem.on('keydown', (e) => {
```

Comment:
Maybe a little concerned that this will not always work as intended (all work completing before watcher has a chance to fire).  Maybe that is unfounded though.

Reply:
It's a good point and something I just assumed. I figured that the digest cycle doesn't kickoff until all interactions are "done", and I assumed that any listeners directly listening to UI interacts will be handled prior to anything angular does, but I could be mistaken there

Reply:
Methods run to completion in JS, so this should be good, just being paranoid.